projects/HieraSeg_project/decode_head/sep_aspp_contrast_head.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import ConvModule, DepthwiseSeparableConvModule, build_norm_layer
+from torch import Tensor
+
+from mmseg.models.decode_heads.aspp_head import ASPPHead
+from mmseg.models.decode_heads.sep_aspp_head import \
+    DepthwiseSeparableASPPModule
+from mmseg.models.losses import accuracy
+from mmseg.models.utils import resize
+from mmseg.registry import MODELS
+
+
+class ProjectionHead(nn.Module):
+
+    def __init__(self, dim_in, norm_cfg, proj_dim=256, proj='convmlp'):
+        super().__init__()
+
+        if proj == 'linear':
+            self.proj = nn.Conv2d(dim_in, proj_dim, kernel_size=1)
+        elif proj == 'convmlp':
+            self.proj = nn.Sequential(
+                nn.Conv2d(dim_in, dim_in, kernel_size=1),
+                build_norm_layer(norm_cfg, dim_in)[1], nn.ReLU(inplace=True),
+                nn.Conv2d(dim_in, proj_dim, kernel_size=1))
+
+    def forward(self, x):
+        return torch.nn.functional.normalize(self.proj(x), p=2, dim=1)
+
+
+@MODELS.register_module()
+class DepthwiseSeparableASPPContrastHead(ASPPHead):
+    """Encoder-Decoder with Atrous Separable Convolution for Semantic Image
+    Segmentation. This head is the implementation of `DeepLabV3+
+
```

Comment:
Might inherit from `mmseg.models.decode_head.DepthwiseSeparableASPPHead` and add a parameter in the type of `dict` to customize the `ProjectionHead`